tracer/src/Datadog.Trace/Util/UriHelpers.cs

```diff
@@ -33,21 +33,46 @@ public static string CleanUri(Uri uri, bool removeScheme, bool tryRemoveIds)
         }
 
         public static string GetCleanUriPath(Uri uri)
-        {
-            return GetCleanUriPath(uri.AbsolutePath);
-        }
+            => GetCleanUriPath(uri.AbsolutePath);
+
+        public static string GetCleanUriPath(Uri uri, string virtualPathToRemove)
+            => GetCleanUriPath(uri.AbsolutePath, virtualPathToRemove);
 
         public static string GetCleanUriPath(string absolutePath)
+            => GetCleanUriPath(absolutePath, null);
+
+        /// <summary>
+        /// Cleans identifiers from an absolute path, and optionally removes the provided prefix
+        /// </summary>
+        /// <param name="absolutePath">The path to clean</param>
+        /// <param name="virtualPathToRemove">The optional virtual path to remove from the front of the path</param>
+        /// <returns>The cleaned path</returns>
+        public static string GetCleanUriPath(string absolutePath, string virtualPathToRemove)
         {
             if (string.IsNullOrWhiteSpace(absolutePath) || (absolutePath.Length == 1 && absolutePath[0] == '/'))
             {
                 return absolutePath;
             }
 
-            // Sanitized url will be at worse as long as the original
-            var sb = StringBuilderCache.Acquire(absolutePath.Length);
+            if (!string.IsNullOrEmpty(virtualPathToRemove) && string.Equals(absolutePath, virtualPathToRemove))
+            {
+                return "/";
+            }
+
+            // If the virtual path is "/" then we're hosted at the root, so nothing to remove
+            // If not, it will be off the form "/myapp", so remove whole thing
+            // Make sure we only remove _whole_ segment i.e. /myapp/controller, but not /myappcontroller
+            var hasPrefix = !string.IsNullOrEmpty(virtualPathToRemove)
+                         && virtualPathToRemove != "/"
+                         && absolutePath.StartsWith(virtualPathToRemove, StringComparison.OrdinalIgnoreCase)
```

Comment:
Should we also assert that `virtualPathToRemove[0] == '/'`?

Reply:
I don't _think_ there's a way for it not to be, but doesn't hurt!